### DISCUSSION TITLE
What the "windowInsetsPadding" methods do ?

### DISCUSSION THREAD

**Discussion by florian-mlr** [android/nowinandroid#296] [Q&A]
Hello, first of all I wanted to thank you for sharing your sample which is very inspiring. 
I'm learning compose and these lines of code regarding padding are blurry to me even after reading the documentation. 😅
```
 Scaffold(....) { padding ->
        Row(
            Modifier.fillMaxSize()
                .windowInsetsPadding(
                    WindowInsets.safeDrawing.only(
                        WindowInsetsSides.Horizontal
                    )
                )
        )
```

```
NavHost(modifier = Modifier
                        .padding(padding)
                        .consumedWindowInsets(padding)

```

Visually what does it bring to the UI ? If you have articles on this subject, I am interested in deepening it.

**Comment by alexvanyo** [comment]
The `windowInsetsPadding` modifiers and related code allow the app to display edge-to-edge, and render underneath the system UI (like the navigation bars and status bars)

https://www.youtube.com/watch?v=mlL6H-s0nF0 goes into more detail about how the API works, and you can see it in action in Now in Android.
